spec/models/pending_induction_submission_batch_spec.rb

```diff
@@ -0,0 +1,142 @@
+
+    describe ".new_claim_for" do
+      subject(:claim) { described_class.new_claim_for(appropriate_body:) }
+
+      it 'creates a batch claim' do
+        expect(claim).to be_a(PendingInductionSubmissionBatch)
+        expect(claim).to be_claim
+        expect(claim).to be_pending
+        expect(claim.appropriate_body).to eq(appropriate_body)
+      end
+    end
+
+    describe ".new_action_for" do
+      subject(:action) { described_class.new_action_for(appropriate_body:) }
+
+      it 'creates a batch action' do
+        expect(action).to be_a(PendingInductionSubmissionBatch)
+        expect(action).to be_action
+        expect(action).to be_pending
+        expect(action.appropriate_body).to eq(appropriate_body)
+      end
+    end
+
+    describe ".build_row_class" do
+      subject(:row_class) { described_class.build_row_class(%i[foo bar baz]) }
+
+      it 'errors if the wrong attributes are used' do
+        expect { row_class.new }.to raise_error('missing keywords: :foo, :bar, :baz')
+      end
+
+      it 'creates an enumerable Data class to encapsulate CSV row cells' do
+        row = row_class.new(foo: 'foo', bar: 'bar', baz: 'baz')
+        expect(row).to be_a(row_class)
+        expect(row).to be_a(Enumerable)
+        expect(row.to_a).to eq(%w[foo bar baz])
+        expect(row).to respond_to(:foo)
+        expect(row).not_to respond_to(:buz)
+      end
+
+      it 'string encodes values' do
+        row = row_class.new(foo: "André", bar: "Zoë", baz: "Ştefan")
+        expect(row.to_a).to eq(%w[André Zoë Ştefan])
+      end
+    end
+  end
+
+  describe "validations" do
+    it { is_expected.to be_valid }
+
+    it { is_expected.to validate_presence_of(:batch_type) }
+    it { is_expected.to allow_value('action').for(:batch_type) }
+    it { is_expected.to allow_value('claim').for(:batch_type) }
+
+    it { is_expected.to validate_presence_of(:batch_status) }
+    it { is_expected.to allow_value('pending').for(:batch_status) }
+    it { is_expected.to allow_value('processing').for(:batch_status) }
+    it { is_expected.to allow_value('processed').for(:batch_status) }
+    it { is_expected.to allow_value('completed').for(:batch_status) }
+    it { is_expected.to allow_value('failed').for(:batch_status) }
+
+    context 'when the attached file is invalid' do
+      subject(:batch) { FactoryBot.build(:pending_induction_submission_batch, :claim, csv_file:) }
+
+      describe "#csv_mime_type" do
+        let(:csv_file) { Rack::Test::UploadedFile.new(Rails.root.join('spec/fixtures/foo.txt'), 'text/plain') }
+
+        specify do
+          expect(batch).not_to be_valid
+          expect(batch.errors[:csv_file]).to include("File type must be a CSV")
+        end
+      end
+
+      describe "#wrong_headers" do # softened rules so order doesn't matter - still TBC
+        let(:csv_file) { Rack::Test::UploadedFile.new(Rails.root.join('spec/fixtures/claims/invalid_missing_columns.csv'), 'text/csv') }
```

Comment:
As the CSVs are only short might it make sense to just set them in a HEREDOC, something like:

```suggestion
        let(:csv_data) do
          <<~CSV
          trn,dob,end_date,number_of_terms,objective,error
          7654321,30-06-1981,2025-01-30,0.5,,
          1234567,30/06/1981,2025-01-30,7.2,,
          CSV
        end
        let(:csv_file) { Rack::Test::UploadedFile.new(csv_data, 'text/csv') }
```

Reply:
I'm already using HEREDOCs in other specs. I retained a mix of testing methods because this way we had the files to hand for manual tests and product review, something other colleagues can open and edit. I think it is definitely something to tidy up after. I have factories and shared specs so we can see what fits best.

Reply:
@peteryates we will eventually get rid of most fixtures once the validation revisions are completed in a follow PR. The parsed CSV data is immediately saved to the DB meaning most specs can just create an array of hashes now.